### ISSUE TITLE
Upgrade to OpenSSL 1.1.0

### ISSUE THREAD

**Issue by ibc** [ibc/mediasoup#49]
There are some API changes to be considered, see https://github.com/cisco/libsrtp/pull/211/files

**Comment by ibc** [comment]
NOTE: Let's wait until Node.js upgrades to openssl 1.1.0.

**Comment by saghul** [comment]
Interesting news: OpenSSL 1.1 no longer needs the application to provide locking facilities! https://www.openssl.org/blog/blog/2017/02/21/threads/

**Comment by ibc** [comment]
Yep, but I won't upgrade to OpenSSL 1.1.0 until Node does it. I just want to drop the `openssl` folder of Node into mediasoup :)
